nautobot/core/management/commands/upgrade.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+
+HELP_TEXT = """
+Performs Nautobot common server upgrade operations using a single entrypoint.
+
+This will run the following management commands with default settings, in order:
+
+- migrate
+- trace_paths
+- collectstatic
+- remove_stale_contenttypes
+- clearsessions
+- invalidate all
+"""
+
+
+class Command(BaseCommand):
+    help = HELP_TEXT
+
+    def create_parser(self, *args, **kwargs):
+        """Custom parser that can display multiline help."""
+        parser = super().create_parser(*args, **kwargs)
+        parser.formatter_class = argparse.RawTextHelpFormatter
+        return parser
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--no-clearsessions",
+            action="store_false",
+            dest="clearsessions",
+            default=True,
+            help="Do not automatically clean out expired sessions.",
+        )
+        parser.add_argument(
+            "--no-collectstatic",
+            action="store_false",
+            dest="collectstatic",
+            default=True,
+            help="Do not automatically collect static files into a single location.",
+        )
+        parser.add_argument(
+            "--no-invalidate-all",
+            action="store_false",
+            dest="invalidate_all",
+            default=True,
+            help="Do not automatically invalidate cache for entire application.",
+        )
+        parser.add_argument(
+            "--no-migrate",
+            action="store_false",
+            dest="migrate",
+            default=True,
+            help="Do not automatically perform any database migrations.",
+        )
+        parser.add_argument(
+            "--no-remove-stale-contenttypes",
+            action="store_false",
+            dest="remove_stale_contenttypes",
+            default=True,
+            help="Do not automatically remove stale content types.",
+        )
+        parser.add_argument(
+            "--no-trace-paths",
```

Comment:
Have we confirmed whether `trace_paths` is even useful/necessary for Nautobot? I get the impression it was originally added in NetBox specifically to support migrations from 2.9 to 2.10.

Reply:
It appears that it may be necessary if objects are removed or manipulated through the ORM, since a portion of the validation is still happening through forms.

There is little cost to running it if it doesn't need to be ran, since ultimately aside from the initial lookup, it ultimately results in a loop.